User/User.php

```diff
@@ -80,4 +88,42 @@ public function __toString()
     {
         return $this->getDisplayName();
     }
+
+    /**
+     * @return self
+     */
+    public function generatePasswordResetToken()
+    {
+        $this->passwordResetToken = PasswordResetToken::getToken();
```

Comment:
zou `PasswordResetToken::generate()` geen logischere method name zijn in je PasswordResetToken object?